Week10/dongyeol02/Mission1/src/hooks/useFetch.ts

```diff
@@ -0,0 +1,33 @@
+import type { AxiosRequestConfig } from "axios";
+import { useEffect, useState } from "react";
+import { axiosClient } from "../pages/apis/axiosClient";
+
+const useFetch = <T>(url: string, options?: AxiosRequestConfig) => {
+  const [data, setData] = useState<string | null>(null);
```

Comment:
현재 useFetch에서 제네릭 `<T>`를 선언해놨지만 실제로는 활용하지 못하고 있는 상황입니다. 
`useState<string | null>(null)`로 하드코딩되어 있어서 `Homepage.tsx`에서 `useFetch<MovieResponse>`로 호출해도 `data`의 타입이 `string`으로 고정되어 버립니다.
    
이로 인해 `data?.results`에 접근할 때 TypeScript가 올바른 타입 추론을 하지 못하고, 런타임에서 예상치 못한 에러가 발생할 위험이 있습니다. 제네릭 타입 `T`를 활용하여 `useState<T | null>(null)`로 수정하면 호출하는 곳에서 지정한 타입(`MovieResponse`)이 제대로 반영되어 타입 안정성이 크게 향상될 것입니다!

```suggestion
const useFetch = <T>(url: string, options?: AxiosRequestConfig) => {
  const [data, setData] = useState<T | null>(null);
```